packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/ELKLayoutedDiagramProvider.java

```diff
@@ -223,6 +230,45 @@ private Label getLayoutedLabel(Label label, Map<String, ElkGraphElement> id2ElkG
         return layoutedLabel;
     }
 
+    private Label getNodeLayoutedLabel(Node node, Map<String, ElkGraphElement> id2ElkGraphElements, double xOffset, double yOffset, ISiriusWebLayoutConfigurator layoutConfigurator) {
+        Label layoutedLabel = node.getLabel();
+        var optionalElkBeginLabel = Optional.of(id2ElkGraphElements.get(node.getLabel().getId().toString())).filter(ElkLabel.class::isInstance).map(ElkLabel.class::cast);
+        if (optionalElkBeginLabel.isPresent()) {
+            ElkLabel elkLabel = optionalElkBeginLabel.get();
+
+            Size size = Size.of(elkLabel.getWidth(), elkLabel.getHeight());
+
+            Position position = Position.at(xOffset + elkLabel.getX(), yOffset + elkLabel.getY());
+            Size nodeSize = Optional.of(elkLabel.getParent()).filter(ElkNode.class::isInstance).map(ElkNode.class::cast).map(elkNode -> {
+                return Size.of(elkNode.getWidth(), elkNode.getHeight());
+            }).get();
```

Comment:
You are calling optional#get without checking first that the optional is not empty

Reply:
done